Sources/Networking/Core/APIManager.swift

```diff
@@ -12,19 +12,19 @@ open class APIManager: APIManaging {
     private let requestAdapters: [RequestAdapting]
     private let responseProcessors: [ResponseProcessing]
     private let errorProcessors: [ErrorProcessing]
-    private let urlSession: URLSession
+    private let responseProvider: ResponseProviding
     private let sessionId: String
     private var retryCounter = Counter()
     
     public init(
-        urlSession: URLSession = URLSession(configuration: .default),
+        responseProvider: ResponseProviding = URLSession(configuration: .default),
```

Comment:
I suggest having 2 initializers. One with the old signature with `urlSession: URLSession` and the new one. They would be almost duplicate but I find the `responseProvider` parameter name extremely misleading if you want to initialize an API manager just for standard use with a URL session